### PR TITLE
Bugfix: handle RegExp in Astro syntax

### PR DESCRIPTION
**Pull request by drwpow**
Tests added! Was investigating our `Code` internal component and realized this broke the compiler:

```
  // Replace "shiki" css variable naming with "astro".
  html = html.replace(/style="(background-)?color: var\(--shiki-/g, 'style="$1color: var(--astro-code-');
```

### CODE REVIEW

File: internal/token.go [hunk 1]
```diff
@@ -791,112 +791,74 @@ func (z *Tokenizer) readUntilCloseAngle() {
...
+	// template literal
 	case '`':
-		z.readTemplateLiteralString()
+		// Note that we DO NOT have to handle `${}` here because our expression
+		// behavior already handles `{}`. Technically incorrect, but it works.
+		z.readUntilChars([]byte{'`'})
 	}
 }
 
-func (z *Tokenizer) readSingleQuoteString() {
```
Comment (drwpow) [inline internal/token.go:803]: Did a bit of cleanup in the tokenizer. We had a bunch of different utilties that all basically performed the same action:

- readSingleQuoteString()
- readDoubleQuoteString()
- readTemplateLiteralString()
- readJSSingleLineComment()
- readJSMultiLineComment()
- readRegExp()

All they did was just keep scanning until a specific character was encountered. That part has just been centralized into:

```
z.readUntilChars([]byte{'\'', '\r', '\n'})
```

This will read until any of these characters are encountered, handles escaped characters, and handles errors a bit more gracefully than the previous methods (hopefully fewer panics).

**Comment by drwpow** [comment]
cc @matthewp I’m going to merge this & publish a new compiler version, but feel free to review post-merge if you’d like!

TL;DR no tests changed, no behavior changed, only a bugfix. All tests pass.
